src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -254,21 +254,38 @@ public override Task SendConnectionsAsync(IReadOnlyList<string> connectionIds, s
     }
 
     /// <inheritdoc />
-    public override Task SendGroupsAsync(IReadOnlyList<string> groupNames, string methodName, object?[] args, CancellationToken cancellationToken = default)
+    public override async Task SendGroupsAsync(IReadOnlyList<string> groupNames, string methodName, object?[] args, CancellationToken cancellationToken = default)
     {
         ArgumentNullException.ThrowIfNull(groupNames);
-        var publishTasks = new List<Task>(groupNames.Count);
-        var payload = _protocol.WriteInvocation(methodName, args);
+        HashSet<string>? connections = null;
 
         foreach (var groupName in groupNames)
         {
             if (!string.IsNullOrEmpty(groupName))
             {
-                publishTasks.Add(PublishAsync(_channels.Group(groupName), payload));
+                var groupChannel = _channels.Group(groupName);
+                if (groupChannel != null)
+                {
+                    var connectionStore = _groups.GetStore(groupChannel);
+                    if (connectionStore != null)
+                    {
+                        foreach (var connection in connectionStore)
+                        {
+                            if (connections == null)
```

Comment:
Can similar optimizations as above be applied here?

Reply:
And to the ToList() call below. I wonder if we shouldn't be using a pooled Span<string> instead.